examples/service_worker/src/lib.rs

```diff
@@ -134,213 +86,232 @@ struct PushSubscriptionKeys {
 //    Update
 // ------ ------
 
-enum ServiceWorkerStatus {
-    Activated,
-    Waiting,
-    StateChange(web_sys::ServiceWorkerState),
-    SubscriptionRetrieved,
-}
-
 enum Msg {
-    CacheCleared,
-    FailedToClearCache(ServiceWorkerError),
-    ClearCache,
-    FailedToRegisterPushManager(ServiceWorkerError),
-    FailedToRegisterServiceWorker(ServiceWorkerError),
-    FailedToUnregisterServiceWorker(ServiceWorkerError),
+    WorkerActivated(web_sys::ServiceWorkerRegistration),
+    SubscriptionReceived(Option<PushSubscription>),
+    WorkerRegistrationFailed(ServiceWorkerError),
+    UnregisterWorker,
+    WorkerUnregistered(Option<ServiceWorkerError>),
     RequestNotificationPermission,
-    ServiceWorkerActivated,
-    SubscriptionRetrieved(PushSubscription),
+    NotificationPermissionRequested(Result<web_sys::NotificationPermission, ServiceWorkerError>),
+    Subscribed(Result<PushSubscription, ServiceWorkerError>),
     SendMessage,
-    SetServiceWorker(web_sys::ServiceWorkerRegistration),
-    StatusUpdate(ServiceWorkerStatus),
-    UnregisterServiceWorker,
-    UnregisteredServiceWorker,
+    ClearCache,
+    CacheCleared(Option<ServiceWorkerError>),
 }
 
+#[derive(Debug)]
```

Comment:
It's just a short enum. I don't see a real reason why the reader should jump between files and lose flow. 
However I understand it would make sense for a larger app or when you expect the enum will be bigger and implement more complex logic.
Also I've moved/removed the enum's implementations so it doesn't make sense to group them in a standalone file anymore.

Reply:
👍